frontend/exporter/src/constant_utils.rs

```diff
@@ -204,76 +192,72 @@ pub(crate) fn scalar_to_constant_expr<'tcx, S: BaseState<'tcx>>(
                     provenance
                 )
             };
-            ConstantExprKind::Borrow(Decorated {
-                ty: ty.sinto(s),
-                span: cspan.clone(),
-                contents: Box::new(ConstantExprKind::GlobalName { id: did.sinto(s) }),
-                hir_id: None,
-                attributes: vec![],
-            })
+            ConstantExprKind::Borrow((ConstantExprKind::GlobalName { id: did.sinto(s) }).decorate(ty.sinto(s), cspan.clone()))
         }
         // A [Scalar] might also be any zero-sized [Adt] or [Tuple] (i.e., unit)
         ty::Tuple(ty) if ty.is_empty() => ConstantExprKind::Tuple { fields: vec![] },
         // It seems we can have ADTs when there is only one variant, and this variant doesn't have any fields.
         ty::Adt(def, _) if let [variant_def] = &def.variants().raw && variant_def.fields.is_empty() => {
             ConstantExprKind::Adt{
                 info: get_variant_information(def, &variant_def.def_id, s),
-                vid: Option::Some(0),
                 fields: vec![],
             }
         },
         _ => span_fatal!(s, span, "Unexpected type {:#?} for scalar {:#?}", ty, scalar),
     };
-    Decorated {
-        ty: ty.sinto(s),
-        span: cspan,
-        contents: Box::new(kind),
-        hir_id: None,
-        attributes: vec![],
-    }
+    kind.decorate(ty.sinto(s), cspan)
 }
 
-/// We evaluate the anonymous constants
```

Comment:
The reason why I had put written this function was to make sure the evaluation policy was consistent across the code base.

Reply:
Makes sense indeed! Just factored out the handling of unevaluated constants in 0514b83b4bd2965e9569a745919eb3a9b492999b